src/allocator_defrag.c

```diff
@@ -349,16 +349,26 @@ unsigned long allocatorDefragGetFragSmallbins(void) {
  *    defragmentation is not necessary as moving regions is guaranteed not to change the fragmentation ratio.
  * 2. If the number of non-full slabs (bin_usage->curr_nonfull_slabs) is less than 2, defragmentation is not performed
  *    because there is no other slab to move regions to.
- * 3. If slab utilization < 'avg utilization'*1.125 [code 1.125 == (1000+UTILIZATION_THRESHOLD_FACTOR_MILLI)/1000]
+ * 3. Defrag if the slab is less than 1/8 full to ensure small slabs get defragmented even when average utilization is low.
```

Comment:
Should this be: ... even when average utilization is *not* low enough to trigger it?

```suggestion
 * 3. Defrag if the slab is less than 1/8 full to ensure small slabs get defragmented even when average utilization is above the threshold.
```

Reply:
Or did I get this wrong?

Reply:
The threshold = 1.125 * average utilization, if we have some items immovable, they actually lower the average, as well as the threshold, so the rest small slabs will be above the threshold and will not be defragged

Reply:
Thanks! I think I got it now.

Reply:
I will keep the comment from @madolson